File: cpp/0409-longest-palindrome.cpp
#include <string>
#include <unordered_set>

class Solution {
    public:
        [[nodiscard]] auto longestPalindrome(const std::string &s) -> int {
            auto non_paired_letters = std::unordered_set<char>{};
            for (const auto c : s) {
                if (non_paired_letters.contains(c)) {
                    non_paired_letters.erase(c);
                } else {
                    non_paired_letters.insert(c);
                }
            }
            return static_cast<int>(
                s.size() - non_paired_letters.size() +
                !non_paired_letters.empty()
            );
        }
};

File: py/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        non_paired_letters: set[str] = set()
        for c in s:
            if c in non_paired_letters:
                non_paired_letters.remove(c)
            else:
                non_paired_letters.add(c)
        return len(s) - len(non_paired_letters) + (len(non_paired_letters) > 0)
